### PR TITLE
Chore: variable names should be snake_cased

### PR DESCRIPTION
**Pull request by znicholasbrown**
Closes: #1314

### CODE REVIEW

File: src/utilities/validation.ts [hunk 1]
```diff
@@ -268,4 +269,10 @@ export const isHandle: ValidationMethodFactory = property => value => {
   }
 
   return `${property} must only contain lowercase letters, numbers, and dashes`
-}
\ No newline at end of file
+}
+
+const SNAKE_CASE_REGEX = /^[a-z0-9]+(?:_[a-z0-9]+)*$/
+
+export const isSnakeCase = (message?: string): ValidationRule<unknown> => (value) => {
```
Comment (pleek91) [inline src/utilities/validation.ts:276]: So part of the reason for validation methods is they can return a string error message. Do we really need to to be customizable? I'd expect this to return the proper error message.

File: src/localization/locale/en.ts [hunk 2]
```diff
@@ -101,6 +101,7 @@ export const en = {
     artifacts: 'Artifacts',
     artifactSearch: 'Search artifacts',
     variablesSearch: 'Search variables',
+    variableMustBeSnakeCase: 'Variable names may only contain letters, numbers, and underscores and may not begin or end with an underscore',
```
Comment (pleek91) [inline src/localization/locale/en.ts:104]: If this was a generic snake case message then the validation rule could pretend the name field.

File: src/utilities/validation.ts (hunk 1 above)
Comment (pleek91) [reply src/utilities/validation.ts:276]: The second argument that's passed from the validation is the name of the field.

**Comment by znicholasbrown** [comment]
Fixed that @pleek91 👍🏻

**Review by pleek91 — APPROVED** [review]

**Pull request merged**